lib/storage/storage.go

```diff
@@ -165,19 +165,26 @@ type Storage struct {
 	isReadOnly atomic.Bool
 }
 
+// OpenOptions optional args for MustOpenStorage
+type OpenOptions struct {
+	Retention       time.Duration
+	MaxHourlySeries int
+	MaxDailySeries  int
+}
+
 // MustOpenStorage opens storage on the given path with the given retentionMsecs.
-func MustOpenStorage(path string, retention time.Duration, maxHourlySeries, maxDailySeries int) *Storage {
+func MustOpenStorage(path string, opts OpenOptions) *Storage {
 	path, err := filepath.Abs(path)
 	if err != nil {
 		logger.Panicf("FATAL: cannot determine absolute path for %q: %s", path, err)
 	}
-	if retention <= 0 || retention > retentionMax {
-		retention = retentionMax
+	if opts.Retention <= 0 || opts.Retention > retentionMax {
+		opts.Retention = retentionMax
```

Comment:
nit:
```suggestion
	retention := opts.Retention
	if retention <= 0 || retention > retentionMax {
		retention = retentionMax
```
this would make code shorter and will cause less changes